agent/engine/docker_task_engine.go

```diff
@@ -490,6 +504,16 @@ func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
 		return nil
 	}
 
+	// Start task that is waitng for the execution role credentials
+	_, ok := engine.managedTasks[task.Arn]
+	if !ok {
+		existingTask.SetDesiredStatus(task.GetDesiredStatus())
+		existingTask.SetExecutionRoleCredentialsID(task.GetExecutionCredentialsID())
```

Comment:
I don't think you need to set this again. Whatever's read from the state file should be good enough